### ISSUE TITLE
How to get panos verion without etag

### ISSUE THREAD

**Issue by M0r41** [noperator/panos-scanner#15]
When i surf the pan web, i can't find etag in header. But got "Expires: Thu, 19 Nov 1981 08:52:00 GMT" and "Strict-Transport-Security: max-age=31536000", can i get version via js or other web pages?

**Comment by k4nfr3** [comment]
Hello, 
Unfortunatly since our publication of this tool, Palo Alto is now stripping the Etag responses. 
This tool will only work on legacy versions.
And as you can see in the version-table.txt file, the returned ETag was equivalent to dates of release of the software that doesn't seem to be related to "Expires: Thu, 19 Nov 1981 08:52:00 GMT"
So to my understanding, no you can't get the version with the etag reply.

**Comment by heheda123123** [comment]
Although login.esp does not have an ETag in response, other static resources, such as /global-protect/portal/images/bg.png, will return the following 
ETag: "657ccc1c-871bb", the ETag algorithm is different, can the time be calculated?

**Comment by k4nfr3** [comment]
a new version of the code will be released soon. Some URIs still have the etag enabled, and will enable version detection.
Working on identifiying 2-3 years of back log of versions. PR will be done soon

**Comment by iseesec** [comment]
expect

**Comment by w00dbury** [comment]
@k4nfr3 any luck with your update and are you able to share your methodology on identifying the back versions? Happy to help.

**Comment by k4nfr3** [comment]
Yes it's done. I need to send pr. The time consuming work is the fingerprinting of a lot of versions

**Comment by k4nfr3** [comment]
I'll send pr tonight. Mostly as it is high topic after today's CVE announcement
